musli/pars/R3BMusliCalPar.h

```diff
@@ -76,15 +79,23 @@ class R3BMusliCalPar : public FairParGenericSet
         int index = (signal - 1) * fNumParamsPosFit + indexpar - 1;
         fPosCalParams->AddAt(val, index);
     }
+    void SetMultHitParams(Float_t val, Int_t signal, Int_t indexpar)
+    {
+        // 1-base

```

Comment:
0-base ??

Reply:
In the file R3BMusliHitPar.h, type and deg should be 1-base and the index 0-base because it is used by a TArrayF.
0-base refers to the index variable like you did for the class R3BLosHitPar.h

    void SetEcorrBetaHitParams(Double_t val, Int_t type, Int_t deg)
    {
        // 0 - based
        int index = (type - 1) * fNumParamsEcorrBetaFit + deg - 1;
        fEcorrBetaHitParams->AddAt(val, index);
    }
    void SetZHitParams(Double_t val, Int_t type, Int_t deg)
    {
        // 0 - based
        int index = (type - 1) * fNumParamsZFit + deg - 1;
        fZHitParams->AddAt(val, type - 1);
    }
Maybe there is a mistake in "fZHitParams->AddAt(val, type - 1);" =>    type-1 --> index ??
Please, could you check if there are mistakes in the classes calling these parameter containers?

In R3BMusliCalPar.h signal and indexpar -> 1-base and index --> 0-base

    void SetEneParams(Float_t val, Int_t signal, Int_t indexpar)
    {
        // 1-base  ---> 0-base
        int index = (signal - 1) * fNumParamsEneFit + indexpar - 1;
        fEneCalParams->AddAt(val, index);
    }
    void SetPosParams(Float_t val, Int_t signal, Int_t indexpar)
    {
        // 1-base  ---> 0-base
        int index = (signal - 1) * fNumParamsPosFit + indexpar - 1;
        fPosCalParams->AddAt(val, index);
    }
Please, check that everything works as expected because these classes were developed during the beam time and the probability of mistakes is high.

In the case of R3BLosHitPar.h, pmt and indexpar should be 1-base but seems like indexpar is 1-base and pmt 0-base (see below)

    void SetLEMatchParams(Float_t val, Int_t pmt, Int_t indexpar)
    {
        // 0-base
        int index = pmt * fNumParamsTamexLE + indexpar - 1;
        fLEMatchParams->AddAt(val, index);
    }
    void SetTEMatchParams(Float_t val, Int_t pmt, Int_t indexpar)
    {
        // 0-base
        int index = pmt * fNumParamsTamexTE + indexpar - 1;
        fTEMatchParams->AddAt(val, index);
    }
If you have time, please, check that this part also works correctly.

You can do it when you wish, but if other collaborators need this code to continue with the analysis I think that you could do these corrections over the current PR.

Reply:
Done, made things consistent with respective parameter finders and data structures. Note that most of these param setters are not called by anything within R3Broot because the parameter finders do not exist, for eg. there is no MusliCal2HitPar, and I just find the parameters by myself.

Please do let me know if there are any more corrections